funcx_endpoint/funcx_endpoint/executors/high_throughput/funcx_worker.py

```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/env python3
+from __future__ import annotations
```

Comment:
Big fan of the cleanup in this file. My only (very mild) concern is that we might be further diverging from the `parsl` interface by introducing these changes - that's definitely unlikely, since the whole `funcx_worker` module is a FuncX invention, but I'd like to know for sure either way. Maybe @yadudoc could weigh in here?

Reply:
I share the concern about how possible it will be to turn back the clock in general and start inheriting capabilities from parsl.

However, I don't think we should let it stand in the way of doing things the Right Way™️.

In particular, I found it necessary to refactor this file in order to make it testable.
Tests can now start with
```python
setup_some_relevant_mocks()
worker = FuncXWorker(...)
```
and then run
```
worker.execute_task(...)
```

That previously wasn't possible, and you'd have to feed crafted messages over a dummy socket to test this behavior.